irc_client/views.py

```diff
@@ -130,6 +147,20 @@ def __init__(self, irc_widget: IrcWidget, server_config: config.ServerConfig):
         self.core.start()
         self.handle_events()
 
+    def open_log_file(self, name: str) -> IO[str] | None:
+        name = re.sub(r"[^A-Za-z0-9-_#]", "", name)
+        (self.irc_widget.log_dir / self.core.host).mkdir(parents=True, exist_ok=True)
+        try:
+            file = (self.irc_widget.log_dir / self.core.host / (name + ".log")).open(
+                "a", encoding="utf-8"
+            )
+        except OSError:
+            traceback.print_exc()
+            return None
+        else:
+            print("*** LOGGING BEGINS", time.asctime(), file=file, flush=True)
+            return file
```

Comment:
Maybe the last two couple lines should be inside `try:`, so you more likely get an error if you can't actually write to file (e.g. disk full).